src/utils_lgpl/io_netcdf/packages/io_netcdf/src/io_ugrid.F90

```diff
@@ -4169,6 +4169,86 @@ function ug_get_contacts_count(ncid, contactids, ncontacts) result(ierr)
 
    end function ug_get_contacts_count
 
+!> Extracts mesh names from the contact attribute and queries their topology dimensions.
+!! The contact attribute has format: "mesh1_name: location1 mesh2_name: location2"
+   function ug_get_contact_mesh_topology_dimensions(ncid, contactids, mesh1_topo_dim, mesh2_topo_dim) result(ierr)
+      use netcdf_utils, only: ncu_get_att
+
+      integer, intent(in) :: ncid !< NetCDF data set id
+      type(t_ug_contacts), intent(in) :: contactids !< Mesh contact set
+      integer, intent(out) :: mesh1_topo_dim !< Topology dimension of first mesh
+      integer, intent(out) :: mesh2_topo_dim !< Topology dimension of second mesh
+      integer :: ierr !< Result status (UG_NOERR if successful)
+   
+      character(len=:), allocatable :: contact_attr, contact_name, error_message
+      character(len=nf90_max_name) :: mesh1_name, mesh2_name, location1, location2, temp
+      integer :: istart, icolon, ispace, mesh1_varid, mesh2_varid
+
+      ierr = UG_NOERR
+
+      !> defaults, in case contacts attribute was incorrectly written we assume old situation
+      mesh1_topo_dim = 1
+      mesh2_topo_dim = 2
+
+      ! Get the contact variable name for better error messages
+      ierr = nf90_inquire_variable(ncid, contactids%varids(cid_contacttopo), name=temp) ! nf90_inquire cannot handle allocatable strings
+      if (ierr /= nf90_noerr) then
+         contact_name = 'unknown_contact'  ! Fallback name
+      else
+         contact_name = trim(temp)
+      end if
+      error_message = ' from mesh topology contacts "'//contact_name//'", assuming defaults.'
+
+      ! Get the contact attribute value
+      ierr = ncu_get_att(ncid, contactids%varids(cid_contacttopo), 'contact', contact_attr)
+      call check_ug_error(ierr, 'Could not read contact attribute'//error_message)
+      if (ierr /= nf90_noerr) return
+
+      ! Parse first mesh name
+      icolon = index(contact_attr, ':')
+      if (icolon == 0) then
+         call check_ug_error(UG_SOMEERR, 'Invalid contact attribute format: missing first colon'//error_message)
+         ierr = UG_SOMEERR
+         return
+      end if
+      mesh1_name = trim(adjustl(contact_attr(1:icolon - 1)))
+
+      ! Skip to second mesh name
+      istart = icolon + 2 ! skip space
```

Comment:
Use +1 and adjustl instead: the space after `:` is not always there/optional. Only the space between first `mesh1:face` and second `mesh2:node` part is optional (and may be multiple spaces).